agent/engine/dependencygraph/graph.go

```diff
@@ -236,74 +253,62 @@ func verifyResourceDependenciesResolved(target *apicontainer.Container, existing
 	return true
 }
 
-func linkCanResolve(target *apicontainer.Container, link *apicontainer.Container) bool {
+func containerOrderingDependenciesCanResolve(target *apicontainer.Container,
+	dependsOnContainer *apicontainer.Container,
+	dependsOnStatus string) bool {
 	targetDesiredStatus := target.GetDesiredStatus()
-	linkDesiredStatus := link.GetDesiredStatus()
-	if targetDesiredStatus == apicontainerstatus.ContainerCreated {
-		// The 'target' container desires to be moved to 'Created' state.
-		// Allow this only if the desired status of the linked container is
-		// 'Created' or if the linked container is in 'steady state'
-		return linkDesiredStatus == apicontainerstatus.ContainerCreated || linkDesiredStatus == link.GetSteadyStateStatus()
-	} else if targetDesiredStatus == target.GetSteadyStateStatus() {
-		// The 'target' container desires to be moved to its 'steady' state.
-		// Allow this only if the linked container is in 'steady state' as well
-		return linkDesiredStatus == link.GetSteadyStateStatus()
-	}
-	log.Errorf("Failed to resolve the desired status of the link [%v] for the target [%v]", link, target)
-	return false
-}
+	dependsOnContainerDesiredStatus := dependsOnContainer.GetDesiredStatus()
 
-func linkIsResolved(target *apicontainer.Container, link *apicontainer.Container) bool {
-	targetDesiredStatus := target.GetDesiredStatus()
-	if targetDesiredStatus == apicontainerstatus.ContainerCreated {
-		// The 'target' container desires to be moved to 'Created' state.
-		// Allow this only if the known status of the linked container is
-		// 'Created' or if the linked container is in 'steady state'
-		linkKnownStatus := link.GetKnownStatus()
-		return linkKnownStatus == apicontainerstatus.ContainerCreated || link.IsKnownSteadyState()
-	} else if targetDesiredStatus == target.GetSteadyStateStatus() {
-		// The 'target' container desires to be moved to its 'steady' state.
-		// Allow this only if the linked container is in 'steady state' as well
-		return link.IsKnownSteadyState()
-	}
-	log.Errorf("Failed to resolve if the link [%v] has been resolved for the target [%v]", link, target)
-	return false
-}
+	switch dependsOnStatus {
```

Comment:
it seems like this block of code gets repeated a few times -- can we extract this into a function?

Reply:
There are minute changes...which is why did not extract that into functions

Reply:
Got it. Wasn't sure if they were similar enough or not